### ISSUE TITLE
Deploy new sites from a specific database

### ISSUE THREAD

**Issue by act2281** [opendevshop/devshop#155]
If we could deploy new sites from a specific database that would be great, that way it's using our theme etc, rather then being a "clean profile". This could mean it deploys always from a specific environment, or we upload & define a specific database for it to always use.

The use case for this is when you make a new environment, a clean install will require either enabling modules, or importing the database. Beyond the start of a new site I don't think there is ever a reason to use a clean database.

**Comment by jonpugh** [comment]
Yes, this is an unfortunate side effect of Aegir. I have a patch going to remove the requirement to run installation for a site to become enabled.

https://www.drupal.org/node/2770077

**Comment by jonpugh** [comment]
https://www.drupal.org/node/2754069

**Comment by act2281** [comment]
You have options for using different installation profiles. Could we have some way that we are able to "build" the installation profile that we want all future sites to use, or at least define it?

That way we can have the site built with exactly what we want in it. Not sure how that works.

**Comment by jonpugh** [comment]
I've finally done it! Prevented aegir from automatically running an install profile on install task: https://www.drupal.org/node/2754069#comment-11760846

With this patch to aegir, we can write code that let's us specify install method: drupal profile, sql import, sql-dump-and-sanitize... anything, once we write the code :)

**Comment by act2281** [comment]
Great! I can't wait to start using that.

**Comment by jonpugh** [comment]
@act2281 We've actually got it working now:

![screenshot from 2017-07-18 16-18-49](https://user-images.githubusercontent.com/106420/28337809-d41fab5a-6bd4-11e7-80c0-51d9aca14051.png)
